### PR TITLE
Use -O0 for tests.

### PR DESCRIPTION
**Pull request by dsharletg**
This speeds up building and running the tests significantly, test_correctness drops from 22 min to 17 min (with -j1). Negligible impact to the other tests. I specifically checked test_performance to make sure the tests itself were unaffected, and also saw small but negligible improvement (maybe ~10 seconds faster out of 3 min).

### CODE REVIEW

**Comment by abadams** [comment]
I'm not comfortable doing this for performance tests. I can easily imagine writing one which benchmarks Halide vs C and forgetting about this issue, (actually the pow test already does this, but powf is not sped up by O3 apparently).

File: Makefile [hunk 1]
```diff
@@ -46,6 +46,7 @@ LLVM_AS = $(LLVM_BINDIR)/llvm-as
 LLVM_NM = $(LLVM_BINDIR)/llvm-nm
 LLVM_CXX_FLAGS = -std=c++11  $(filter-out -O% -g -fomit-frame-pointer -pedantic -W% -W, $(shell $(LLVM_CONFIG) --cxxflags | sed -e 's/\\/\//g' -e 's/\([a-zA-Z]\):/\/\1/g;s/-D/ -D/g;s/-O/ -O/g'))
 OPTIMIZE ?= -O3
+OPTIMIZE_FOR_BUILD_TIME ?= -O0
```
Comment (steven-johnson) [inline Makefile]: Do we want to make a similar fix for CMake?

Comment (dsharletg) [reply Makefile]: As far as I can tell, we aren't setting the optimization level for tests at all in the CMake build, except for external_code_extern.cpp...

Comment (steven-johnson) [reply Makefile]: ...which implies the performance tests are dubious for CMake. Opened https://github.com/halide/Halide/issues/2566

**Comment by steven-johnson** [comment]
LGTM pending buildbots.

**Comment by dsharletg** [comment]
Bah the change windows is demanding is viral. I'm going to try again to fix the vtable linker issue on gcc and then probably give up on this change if I can't find a workaround.

**Comment by steven-johnson** [comment]
The change to make EXPORT work on gcc/clang was well-intentioned but is clearly biting us; I'm tempted now to say we should back that out and go back to having EXPORT a no-op everywhere but Windows.

And/or: CMake now has WINDOWS_EXPORT_ALL_SYMBOLS, which purports to be a way to obviate most of that nonsense on Windows and simulate the unixy default of export-everything; it's probably worth trying out to see if it actually works. (https://github.com/halide/Halide/issues/2571)

**Comment by dsharletg** [comment]
Closing this, hopefully #2574 works better.

**Pull request closed without merging**